src/transformers/trainer.py

```diff
@@ -1565,6 +1569,8 @@ def _inner_training_loop(
     ):
         self.accelerator.free_memory()
         self._train_batch_size = batch_size
+        if self.args.auto_find_batch_size:
+            self.state.train_batch_size = self._train_batch_size
```

Comment:
Are we guaranteed that `self.state` exists here?

Reply:
Yes, `state` gets made as part of the `__init__`